packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -97,5 +100,36 @@ Future<void> buildMacOS({
   if (result != 0) {
     throwToolExit('Build process failed');
   }
+  if (buildInfo.codeSizeDirectory != null && sizeAnalyzer != null) {
+    final String arch = getNameForDarwinArch(DarwinArch.x86_64);
+    final File aotSnapshot = globals.fs.directory(buildInfo.codeSizeDirectory)
+      .childFile('snapshot.$arch.json');
+    final File precompilerTrace = globals.fs.directory(buildInfo.codeSizeDirectory)
+      .childFile('trace.$arch.json');
+
+    // This analysis is only supported for release builds.
```

Comment:
(probably all handled in the same place, but) ditto here

Reply:
We check that here: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command.dart#L691